WordPress/Classes/ViewRelated/NUX/Signin2FAViewController.swift

```diff
@@ -0,0 +1,248 @@
+///
+@objc class Signin2FAViewController : NUXAbstractViewController, SigninWPComSyncHandler, SigninKeyboardResponder
+{
+
+    @IBOutlet weak var verificationCodeField: UITextField!
+    @IBOutlet weak var sendCodeButton: UIButton!
+    @IBOutlet weak var submitButton: NUXSubmitButton!
+    @IBOutlet weak var statusLabel: UILabel!
+    @IBOutlet var bottomContentConstraint: NSLayoutConstraint!
+    @IBOutlet var verticalCenterConstraint: NSLayoutConstraint!
+
+    lazy private var loginFacade: LoginFacade = {
+        let facade = LoginFacade()
+        facade.delegate = self
+        return facade
+    }()
+
+
+    /// A convenience method for obtaining an instance of the controller from a storyboard.
+    ///
+    /// - Parameters:
+    ///     - loginFields: A LoginFields instance containing any prefilled credentials.
+    ///
+    class func controller(loginFields: LoginFields) -> Signin2FAViewController {
+        let storyboard = UIStoryboard(name: "Signin", bundle: NSBundle.mainBundle())
+        let controller = storyboard.instantiateViewControllerWithIdentifier("Signin2FAViewController") as! Signin2FAViewController
+        controller.loginFields = loginFields
+        return controller
+    }
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureSendCodeButtonText()
+        configureStatusLabel("")
+        configureSubmitButton(false)
+    }
+
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+
+        configureViewForEditingIfNeeded()
+    }
+
+
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+
+        registerForKeyboardEvents(#selector(SigninEmailViewController.handleKeyboardWillShow(_:)),
+                                  keyboardWillHideAction: #selector(SigninEmailViewController.handleKeyboardWillHide(_:)))
+
+    }
+
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        unregisterForKeyboardEvents()
+
+        // Multifactor codes are time sensitive, so clear the stored code if the
+        // user dismisses the view. They'll need to reentered it upon return. 
+        loginFields.multifactorCode = ""
+        verificationCodeField.text = ""
+    }
+
+
+    // MARK: Configuration Methods
+
+
+    /// Configures the appearance of the button to request a 2fa code be sent via SMS.
+    ///
+    func configureSendCodeButtonText() {
+        // Text: Verification Code SMS
+        let string = NSLocalizedString("Enter the code on your authenticator app or <u>send the code via text message</u>.",
+                                       comment: "Message displayed when a verification code is needed")
+        let options = [
+            NSDocumentTypeDocumentAttribute: NSHTMLTextDocumentType
+        ]
+
+        let styledString = "<style>body {font-family: sans-serif; font-size:14px; color: #ffffff; text-align:center;}</style>" + string
```

Comment:
I've just been doing this same thing for a footer in Plans that has some highlighting. It looks like a font of `sans-serif` uses Helvetica instead of San Francisco. You'll need to use a value of `-apple-system` here if you want SF.